### PR TITLE
add e2e test for service entry with inlined workload entry

### LINKED ISSUE #1037: ServiceEntry doesn't work in Kmesh
**What happened**:

I deployed a ServiceEntry like this :
```yaml
apiVersion: networking.istio.io/v1beta1
kind: ServiceEntry
metadata:
  name: test-se-v4
spec:
  hosts:
  - dummy-v4.example.com
  addresses:
  - 240.240.240.255
  ports:
  - number: 80
    name: http
    protocol: HTTP
  resolution: STATIC
  location: MESH_INTERNAL
  endpoints:
  # Expose bookinfo through gateway first.
  - address: 172.18.0.0
    ports:
      http: 80
```

`172.18.0.0` is the external IP of ingress gateway and I expose `bookinfo` in port `80`, but it doesn't work as expected.

```bash
kubectl exec deploy/sleep -- curl -v http://240.240.240.255:80/productpage
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 240.240.240.255:80...
  0     0    0     0    0     0      0      0 --:--:--  0:00:26 --:--:--     0
```

**What you expected to happen**:

Should be able to access.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kmesh version:
- Others:

### PR DESCRIPTION
**Pull request by YaoZengzeng**
**What type of PR is this?**

/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #1037

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controller/workload/workload_processor.go [hunk 1]
```diff
@@ -609,6 +609,10 @@ func (p *Processor) storeServiceData(serviceName string, waypoint *workloadapi.G
 		newValue.ServicePort[i] = nets.ConvertPortToBigEndian(port.ServicePort)
 		if strings.Contains(serviceName, "waypoint") {
 			newValue.TargetPort[i] = nets.ConvertPortToBigEndian(KmeshWaypointPort)
+		} else if port.TargetPort == 0 {
+			// NOTE: Target port could be unset in servicen entry, in which case it should
+			// be consistent with the Service Port.
+			newValue.TargetPort[i] = nets.ConvertPortToBigEndian(port.ServicePort)
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/workload_processor.go:615]: I can see istio has set the traget port

(https://github.com/istio/istio/blob/081a061e195f6538df2b1dbb610217aa3fe596b6/pilot/pkg/serviceregistry/kube/controller/ambient/services.go#L145-L153)


Which istio version are you using

Comment (hzxuzhonghu) [reply pkg/controller/workload/workload_processor.go:615]: Looks this is set since istio 1.23

**Review by hzxuzhonghu — APPROVED** [review]

File: pkg/controller/workload/workload_processor.go (hunk 1 above)
Comment (YaoZengzeng) [reply pkg/controller/workload/workload_processor.go:615]: oh, I tested it with istio 1.22

**Pull request merged**